cmd/conduit/list.go

```diff
@@ -0,0 +1,214 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/algorand/indexer/exporters"
+	"github.com/algorand/indexer/importers"
+	"github.com/algorand/indexer/processors"
+)
+
+func makeListCmd() *cobra.Command {
+
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "lists all plugins available to conduit",
+		Long:  "lists all plugins available to conduit",
+		Args:  cobra.MaximumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runListCmd(os.Stdout, args)
+		},
+		SilenceUsage: true,
+		// Silence errors because our logger will catch and print any errors
+		SilenceErrors: true,
+	}
+
+	return cmd
+}
+
+func runListCmd(out io.Writer, pluginNames []string) error {
+	if len(pluginNames) == 0 {
+		output, err := outputAll()
+		if err != nil {
+			return err
+		}
+
+		fmt.Fprint(out, output)
+	}
+
+	var potentialNames []string
+	var sampleConfig string
+
+	for _, pluginName := range pluginNames {
+
+		if strings.HasPrefix(pluginName, "importer.") {
```

Comment:
It might be possible to embed a separate `Metadata()` interface in each of the plugin types and use a helper function to print the list of each plugin type.